### PR TITLE
Update getEmbedding to validate config before sending to worker

### PR DESCRIPTION
**Pull request by cosa65**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-551?atlOrigin=eyJpIjoiZTQyY2U5NTA5MDE5NDM2Mzg5Mzk2N2M1YWJhMzUyYjgiLCJwIjoiaiJ9

#### Link to staging deployment URL 
https://ui-nxkczhpg46grz3cnpejga3qy8a.scp-staging.biomage.net

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [x] Unit tests written
- [x] Tested locally with Inframock
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: tests/api/event-services/work-request.test.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ describe('handleWorkRequest', () => {
       socketId: '6789',
       experimentId: 'my-experiment',
       timeout: '2001-01-01T00:00:00Z',
-      body: { name: 'GetEmbedding', config: {} },
+      body: { name: 'GetEmbedding', config: { minimumDistance: 0.3, distanceMetric: 'euclidean' } },
```
Comment (ogibson) [inline tests/api/event-services/work-request.test.js]: The type parameter is missing on these requests, and it should be "tsne/umap/pca"

**Review by ogibson — DISMISSED** [review]
The parameters that need to be checked depend on the type of embedding that's being calculated:
umap:
minimumDistance floating point number (0 - 1)  It's a gross margin beyond usage limits but won't actually break anything within those boundaries. 
distanceMetric (euclidean, cosine)

tsne:
perplexity: default 30; range 5 to (number of cell)/100
learnrate: default value maximum (200 or (number of cell)/12); range 10 to 1000. 

pca:
Let's leave it without parameters for now because it's not implemented in the UI. Just add it to the type enum and on the worker side it'll have hardcoded parameters.

File: tests/api/event-services/work-request.test.js (hunk 1 above)
Comment (marcellp) [reply tests/api/event-services/work-request.test.js]: Can you add it as a parameter, or is the test relying on this not being set?

**Review by marcellp — COMMENTED** [review]
Other than this comment, LGTM.

File: tests/api/event-services/work-request.test.js (hunk 1 above)
Comment (marcellp) [reply tests/api/event-services/work-request.test.js]: Good catch @ogibson-biomage, there is an error in the schema that makes `type` not required.

File: src/specs/models/work-request-bodies/WorkRequestGetEmbedding.v1.yaml [hunk 2]
```diff
@@ -10,6 +10,20 @@ properties:
...
+      distanceMetric:
+        type: string
+        description: The selected method used to measure distance
+        enum:
+          - euclidean
+          - cosine
+    required:
+      - minimumDistance
+      - distanceMetric
   type:
```
Comment (marcellp) [inline src/specs/models/work-request-bodies/WorkRequestGetEmbedding.v1.yaml]: Type should not be deprecated and be required.

**Review by marcellp — APPROVED** [review]

**Pull request merged**